app/controllers/users/registrations_controller.rb

```diff
@@ -50,4 +51,12 @@ def after_sign_up_path_for(resource)
   def after_inactive_sign_up_path_for(resource)
     super(resource)
   end
+
+  def send_signup_emails
+    if @user.persisted?
```

Comment:
try w/
```
return if @user.new_record?
SignupMailer.with(user: @user).welcome_email.deliver_now
SignupMailer.with(user: @user....
```

Reply:
It works.
Why is is better than .persisted?

Reply:
Ok, i think i understood. Now, if object is not saved in db we exit from the function with return and don't send emails.